### ISSUE TITLE
Local entiteis pagination is broken

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#945]
Dicussed with @anabelengp, somehow related with https://github.com/telefonicaid/fiware-orion/issues/847

CB pagination doesn't work for local entities in the case some registration "shadows" the local entity. E.g. 

```
> db.registrations.find();
{ "_id" : ObjectId("556c158b3a1b8459f4f1ad32"), "expiration" : NumberLong(1433147445), "servicePath" : "/Basuras", "format" : "JSON", "contextRegistration" : [     {     "entities" : [     {  "id" : "device_de_ana",     "type" : "device" } ],     "attrs" : [     {     "name" : "PING",     "type" : "command",     "isDomain" : "false" } ],     "providingApplication" : "http://10.0.0.63:8080/iot/ngsi/d" } ] }
{ "_id" : ObjectId("556c18343a1b8459f4f1ad33"), "expiration" : NumberLong(1433233844), "servicePath" : "/Basuras", "format" : "JSON", "contextRegistration" : [     {     "entities" : [     {  "id" : "device_de_ana",     "type" : "device" } ],     "attrs" : [     {     "name" : "PING",     "type" : "command",     "isDomain" : "false" } ],     "providingApplication" : "http://10.0.0.63:8080/iot/ngsi/d" } ] }
```

```
> db.entities.find();
{ "_id" : { "id" : "entity_local", "type" : "local", "servicePath" : "/Basuras" }, "attrNames" : [  "a" ], "attrs" : { "a" : { "type" : "a", "creDate" : 1432296515, "modDate" : 1432296515, "value" : "a" } }, "creDate" : 1432296515, "modDate" : 1432296515 }
{ "_id" : { "id" : "qa_entity2", "servicePath" : "/Electricidad" }, "attrNames" : [ ], "attrs" : {  }, "creDate" : 1432632637, "modDate" : 1432632637 }
{ "_id" : { "id" : "device_de_ana", "type" : "device", "servicePath" : "/Basuras" }, "attrNames" : [  "PING_status",  "TimeInstant", "Brand",  "PING_info" ], "attrs" : { "Brand" : { "value" : "ACME", "type" : "string", "md" : [     {     "name" : "TimeInstant", "type" : "ISO8601",     "value" : "2015-06-01T08:50:03.451905Z" } ], "creDate" : 1433148595, "modDate" : 1433148602 }, "PING_info" : { "value" : "ana@ping|MADE_OK", "type" : "string", "md" : [     { "name" : "TimeInstant", "type" : "ISO8601",     "value" : "2015-06-01T08:50:03.451100Z" } ], "creDate" : 1433148602, "modDate" : 1433148602 }, "PING_status" : { "value" : "OK", "type" : "string", "md" : [     {     "name" : "TimeInstant",     "type" : "ISO8601",     "value" : "2015-06-01T08:50:03.450536Z" } ], "creDate" : 1433148595, "modDate" : 1433148602 }, "TimeInstant" : { "value" : "2015-06-01T08:50:03.451322Z", "type" : "ISO8601", "creDate" : 1433148595, "modDate" : 1433148602 } }, "creDate" : 1433148595, "modDate" : 1433148602 }
```

Pagination request that doesn't work ("Basuras" have devices):

```
curl 'http://10.95.213.244:10026/v1/contextEntities?details=on&limit=15&offset=0' -H 'Fiware-Service: smartvalencia' -H 'Fiware-ServicePath: /Basuras' -H 'Accept: application/json'
{
  "contextResponses" : [
    {
      "contextElement" : {
        "type" : "local",
        "isPattern" : "false",
        "id" : "entity_local",
        "attributes" : [
          {
            "name" : "a",
            "type" : "a",
            "value" : "a"
          }
        ]
      },
      "statusCode" : {
        "code" : "200",
        "reasonPhrase" : "OK"
      }
    },
    {
      "contextElement" : {
        "type" : "device",
        "isPattern" : "false",
        "id" : "device_de_ana",
        "attributes" : [
          {
            "name" : "Brand",
            "type" : "string",
            "value" : "ACME",
            "metadatas" : [
              {
                "name" : "TimeInstant",
                "type" : "ISO8601",
                "value" : "2015-06-01T08:50:03.451905Z"
              }
            ]
          },
          {
            "name" : "PING_info",
            "type" : "string",
            "value" : "ana@ping|MADE_OK",
            "metadatas" : [
              {
                "name" : "TimeInstant",
                "type" : "ISO8601",
                "value" : "2015-06-01T08:50:03.451100Z"
              }
            ]
          },
          {
            "name" : "PING_status",
            "type" : "string",
            "value" : "OK",
            "metadatas" : [
              {
                "name" : "TimeInstant",
                "type" : "ISO8601",
                "value" : "2015-06-01T08:50:03.450536Z"
              }
            ]
          },
          {
            "name" : "TimeInstant",
            "type" : "ISO8601",
            "value" : "2015-06-01T08:50:03.451322Z"
          }
        ]
      },
      "statusCode" : {
        "code" : "200",
        "reasonPhrase" : "OK"
      }
    }
  ]
}
```

Pagination request that works ("Electricidad" doesn't have devices):

```
curl 'http://10.95.213.244:10026/v1/contextEntities?details=on&limit=15&offset=0' -H 'Fiware-Service: smartvalencia' -H 'Fiware-ServicePath: /Electricidad' -H 'Accept: application/json'
{
  "contextResponses" : [
    {
      "contextElement" : {
        "type" : "",
        "isPattern" : "false",
        "id" : "qa_entity2"
      },
      "statusCode" : {
        "code" : "200",
        "reasonPhrase" : "OK"
      }
    }
  ],
  "errorCode" : {
    "code" : "200",
    "reasonPhrase" : "OK",
    "details" : "Count: 1"
  }
} 
```

**Comment by kzangeli** [comment]
"Doesn't work" only means "No **count** in *ErrorCode:;details* in the response-payload, right?
Actually, no *ErrorCode* at all ...
Does the pagination work apart from this error?

**Comment by fgalan** [comment]
> "Doesn't work" only means "No count in ErrorCode:;details in the response-payload, right?
> Actually, no ErrorCode at all ...
> Does the pagination work apart from this error?

Good point. This question is actually for @anabelengp.

**Comment by anabelengp** [comment]
I didn't know but I have checked it and it seems that pagination doesn't work too if CPr have registered attributes

**Comment by fgalan** [comment]
Fixed in PR #948 (develop) and PR #949 (release/0.22.0).
